src/miniscript/types/extra_props.rs

```diff
@@ -25,6 +27,13 @@ pub struct ExtData {
     pub ops_count_sat: Option<usize>,
     /// The worst case ops-count for dissatisfying this Miniscript fragment.
     pub ops_count_nsat: Option<usize>,
+    /// Whether any satisfaction of this fragment contains any timelocks
+    /// contains_timelock[0] = csv with heights
+    /// contains_timelock[1] = csv with times
+    /// contains_timelock[2] = cltv with heights
+    /// contains_timelock[3] = cltv with times
+    /// contains_timelock[4] = combination of any heightlocks and timelocks
```

Comment:
Please create a structure for this instead of using a `[bool; 5]` :)

Reply:
You can impl `Default` on it so you can construct it quickly

Reply:
To be clear, I can't review this PR any further until this is fixed :)

Reply:
@apoelstra, This meaning the logic condition? I can change that quickly